lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +252,23 @@ def update_state_file
     end
   end
 
+  private
+
+  def enable_encoding
+    !@charset.nil? || !@columns_charset.empty?
+  end
+
+  # make sure the encoding is uniform over fields
+  def convert(column_name, value)
+    return value unless value.is_a?(String)
+    if @columns_charset[column_name]
+      converter = @converters[@columns_charset[column_name]]
+      converter.convert(value)
+    elsif @charset
+      converter = @converters[@charset]
+      converter.convert(value)
+    else
+      value
+    end
+  end
```

Comment:
well, `if (column_name_charset = @columns_charset[column_name])` might be error prone but if there is no error, there's no error and it's not like an error will magically happen. plus if it is well tested then what can go wrong? 

if you are not comfortable with this style then maybe if you prefer you can break it in 2 lines? again, this is just a simple optimization to avoid useless double hash lookup for the same key.
```ruby
column_name_charset = @columns_charset[column_name]
if column_name_charset
```